android/Idviu/app/src/main/java/com/idviu/sample/hss/HSSPlayerActivity.java

```diff
@@ -0,0 +1,268 @@
+import com.labgency.player.LgyTrack;
+import com.labgency.player.LgyTrack.TrackType;
+import com.idviu.sample.hss.HSSSampleApplication.HSSCustomLicenseRequestHandler;
+
+import io.streamroot.dna.core.BandwidthListener;
+import io.streamroot.dna.core.DnaClient;
+import io.streamroot.dna.core.PlayerInteractor;
+import io.streamroot.dna.core.QosModule;
+import io.streamroot.dna.utils.stats.StatsView;
+import io.streamroot.dna.utils.stats.StreamStatsManager;
+
+/**
+ * Sample Player Activity using HSSPlayer
+ */
+@SuppressLint("NewApi")
+public class HSSPlayerActivity extends FragmentActivity implements OnCloseEventListener, OnErrorListener, OnCompletionListener, AdaptiveStreamingListener {
+
+    private static final String TAG = "HSSPlayerActivity";
+
+    private ProgressBar mProgress = null;
+    private RelativeLayout mRootView;
+	private HSSPlayerView mPlayerView = null;
+	private DnaClient dnaClient;
+	private StreamStatsManager streamStatsManager;
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB){
+			getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+		}
+		getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+		
+		setContentView(R.layout.player);
+		mPlayerView = (HSSPlayerView) findViewById(R.id.playerView);
+		mPlayerView.setRetainedNonInstanceStateObject(getLastCustomNonConfigurationInstance());
+		mPlayerView.setOnCompletionListener(this);
+		mPlayerView.showDebugInfos(true);
+		mPlayerView.setOnErrorListener(this);
+		mPlayerView.setAdaptiveStreamingListener(this);
+		mPlayerView.setOnCloseEventListener(this); //set this listener to show the "close" button and be able to stop playback
+		mPlayerView.setPlayerParam("connect_timeout", "10000"); //only 10s for connection timeout, reduces the risk of HTTP 403 for some HLS streams with temporary sessions
+		mPlayerView.setPlayerParam("start_low_profile", "1");
+		mPlayerView.setWidevineOfflineMode(false); //set to true if you want to retrieve a persistent license for Widevine
+		mPlayerView.setOnPreparedListener(new OnPreparedListener() {
+
+			@Override
+			public void onPrepared(MediaPlayer mp) {
+				Log.i(TAG, "redirected url: " + mPlayerView.getPlayer().getRedirectedUrl());
+			}
+		});
+		HSSCustomLicenseRequestHandler.instance().setPlayerView(mPlayerView);
+
+		Intent intent = this.getIntent();
+		Bundle bundle = intent.getExtras();
+		Entry entry = (Entry)bundle.getSerializable("entry");
+		if (mPlayerView.getPlaylist().getSize() == 0 && entry != null ){ //only if empty
+			HSSPlaylistItem item = null;
+				if (entry.downloadId >= 0){
+					item = HSSPlaylistItem.createWithDownloadId(entry.downloadId);
+					if (entry.downloadForceLocalMode)
+						item.setForceLocalMode(entry.downloadForceLocalMode);
+					mPlayerView.getPlaylist().addItem(item);
+				}else {
+					if (entry.streamrootDnaEnabled) {
+						// streamroot init
```

Comment:
I think as an integration example maybe we should somehow "highlight" the differences compared to normal way of using the player, especially from what I see there's some Android UI stuffs going on too.